core/domain/blog_services.py

```diff
@@ -798,20 +821,21 @@ def get_published_blog_post_summaries(
         max_limit = size
     else:
         max_limit = feconf.MAX_NUM_CARDS_TO_DISPLAY_ON_BLOG_HOMEPAGE
-    blog_post_rights_models: Sequence[blog_models.BlogPostRightsModel] = (
-        blog_models.BlogPostRightsModel.query(
-            blog_models.BlogPostRightsModel.blog_post_is_published == True  # pylint: disable=singleton-comparison
+    # We use '!= None' instead of 'is not None' because the it is inside the
+    # query() as GAE does not support 'is not None' inside the query().
+    # Using the latter results in the following error:
+    # 'Cannot filter a non-Node argument'.
```

Comment:
```suggestion
    # We use '!= None' instead of 'is not None' because it is used inside
    # 'query()' and Google App Engine does not support 'is not None' 
    # inside 'query()'.
```

Reply:
Done